enzyme/Enzyme/MLIR/Analysis/ActivityAnnotations.cpp

```diff
@@ -0,0 +1,1319 @@
+#include "ActivityAnnotations.h"
+#include "DataFlowAliasAnalysis.h"
+#include "DataFlowLattice.h"
+#include "Dialect/Dialect.h"
+#include "Dialect/Ops.h"
+
+#include "mlir/Analysis/DataFlow/ConstantPropagationAnalysis.h"
+#include "mlir/Analysis/DataFlow/DeadCodeAnalysis.h"
+#include "mlir/Analysis/DataFlow/DenseAnalysis.h"
+#include "mlir/Analysis/DataFlow/SparseAnalysis.h"
+#include "mlir/Analysis/DataFlowFramework.h"
+#include "mlir/Interfaces/FunctionInterfaces.h"
+#include "llvm/Support/raw_ostream.h"
+
+// TODO: Remove dependency on dialects in favour of differential dependency
+// interface
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+
+using namespace mlir;
+
+static bool isPossiblyActive(Type type) {
+  return isa<FloatType, ComplexType>(type);
+}
+
+template <typename ValueT>
+void printSetLattice(const enzyme::SparseSetLattice<ValueT> &setLattice,
+                     raw_ostream &os) {
+  if (setLattice.isUnknown()) {
+    os << "Unknown Origin";
+  } else if (setLattice.isUndefined()) {
+    os << "Undefined Origin";
+  } else {
+    os << "size: " << setLattice.getElements().size() << ":\n";
+    for (auto element : setLattice.getElements()) {
+      os << "  " << element << "\n";
```

Comment:
Nit: it would be helpful to have these sorted in a consistent way, e.g., by printing each element to a string via `raw_string_ostream` first, then sorting a vector of strings before printing them to `os`.

Reply:
I realized that the consistent order via sorting was already implemented for the `serialize` method. The `print` method isn't actually used for any tests (it's just required by the framework), so we can re-use the `serialize` method and print out the serialized Attributes rather than duplicating the sorting logic.